src/editorial/web/components/button/index.tsx

```diff
@@ -0,0 +1,266 @@
+import { css } from '@emotion/react';
+
+import { Design, Display, Format, Pillar, Special } from '@guardian/types';
+import {
+	Button as CoreButton,
+	ButtonProps as CoreButtonProps,
+	LinkButton as CoreLinkButton,
+	LinkButtonProps as CoreLinkButtonProps,
+	ButtonPriority,
+} from '@guardian/src-button';
+import {
+	culture,
+	labs,
+	lifestyle,
+	news,
+	opinion,
+	specialReport,
+	sport,
+	neutral,
+} from '@guardian/src-foundations';
+
+export interface ButtonProps extends CoreButtonProps {
+	format?: Format;
+	priority?: ButtonPriority;
+	children: React.ReactNode;
+}
+
+export interface LinkButtonProps extends CoreLinkButtonProps {
+	format?: Format;
+	priority?: ButtonPriority;
+	children: React.ReactNode;
+}
+
+const defaultFormat = {
+	display: Display.Standard,
+	design: Design.Article,
+	theme: Pillar.News,
+};
+
+const WHITE = neutral[100];
+
+const decideBackground = (format: Format, priority: ButtonPriority) => {
+	switch (priority) {
+		case 'primary':
+		case 'secondary':
+			switch (format.theme) {
+				case Pillar.News:
+					return css`
+						background-color: ${news[300]};
```

Comment:
I would like to see these colours expressed in a more context-aware way (e.g. `background.ctaPrimaryNews` as demonstrated in the [Tokens documentation](https://theguardian.design/2a1e5182b/p/1377a6-tokens/b/293ddb)). I know we are still talking about how the Editorial palette will be structured, so perhaps we can come back to this when the structure is clearer.

Reply:
In DCR, we use the `news`, `opinion`, etc. objects everywhere. We vertually never use any of the tokens. This is for two reasons

1. The tokens often don't work for our use cases, they seem very banner/callout orientated, but also mainly because
2. The design team specifically gives use `news[300]` etc., as the required colours to use 🤷